corfudb-tools/src/main/java/org/corfudb/browser/CorfuStoreBrowser.java

```diff
@@ -133,11 +138,18 @@ public int printTable(String namespace, String tablename) {
                 Iterables.partition(entryStream::iterator, batchSize);
         for (List<Map.Entry<CorfuDynamicKey, CorfuDynamicRecord>> partition : partitions) {
             for (Map.Entry<CorfuDynamicKey, CorfuDynamicRecord> entry : partition) {
-                builder = new StringBuilder("\nKey:\n" + entry.getKey().getKey())
-                        .append("\nPayload:\n").append(entry.getValue().getPayload())
-                        .append("\nMetadata:\n").append(entry.getValue().getMetadata())
-                        .append("\n====================\n");
-                log.info(builder.toString());
+                try {
+                    builder = new StringBuilder("\nKey:\n")
+                            .append(JsonFormat.printer().print(entry.getKey().getKey()))
+                            .append("\nPayload:\n")
+                            .append(JsonFormat.printer().print(entry.getValue().getPayload()))
+                            .append("\nMetadata:\n")
+                            .append(JsonFormat.printer().print(entry.getValue().getMetadata()))
+                            .append("\n====================\n");
+                    log.info(builder.toString());
+                } catch (InvalidProtocolBufferException e) {
+                    log.error("invalid protobuf: ", e);
```

Comment:
can we print the protobuf type which caused the exception?

Reply:
in this case there is only one protobuf - the TableName, also I think the error would carry the line where it failed. if it does.

Reply:
sorry, but arent the keys and values the protobufs?

Reply:
yes same protobuf TableName is being used as key and also as value